### ISSUE TITLE
Invalid version reported when installed without .git

### ISSUE THREAD

**Issue by sanmai** [maglnet/ComposerRequireChecker#109]
Install like so:
```
composer require --prefer-dist --dev maglnet/composer-require-checker
```
(`--prefer-dist` [is the default](https://getcomposer.org/doc/03-cli.md#install-i), but just to make sure...)

Observe:
```
$ ./vendor/bin/composer-require-checker check composer.json
ComposerRequireChecker v3.0.1-15-ge5fafe9-dev
...
```
But that's exactly my package's last version:
```
$ git describe --tags --dirty=-dev --always
v3.0.1-15-1beef234-dev
```
One way to tackle this problem is to [use Git's `export-subst` trick](https://stackoverflow.com/a/55699582/93540)  for, say `Cli/Application.php`.

 [Quick demo.](https://3v4l.org/XD2q3)

**Comment by sanmai** [comment]
One way to tackle this problem is to [use `export-subst` for a file](https://git-scm.com/docs/gitattributes#_code_export_subst_code), say `Cli/Application.php`, then adding the following constant:

```php
const VERSION_INFO = '$Format:%h%d$';
```
```sh
$ grep export-subst .gitattributes
/src/ComposerRequireChecker/Cli/Application.php        export-subst
```
Then, on export, this constant will be replaced with the following:
```sh
$ git tag -m "Releasing 2.0.1" -a 2.0.1
$ git archive 2.0.1 | grep -a VERSION_INFO
    const VERSION_INFO = '36ecb45 (HEAD -> master, tag: 2.0.1)';
```
Next you can check if `self::VERSION_INFO` begins with a dollar sign, and if not-parse out the version and the commit ID with a simple regular expression.

It may not be include branch name though:
```sh
git log --pretty=format:%h%d
```

**Comment by Ocramius** [comment]
Maybe we should just use `ocramius/package-versions` there

**Comment by sanmai** [comment]
If it is acceptable to have another dependency, it'll work just as well if not better.

**Comment by Lctrs** [comment]
Fixed by #146.

**Comment by Ocramius** [comment]
Thanks @Lctrs!
